concent_api/common/decorators.py

```diff
@@ -85,3 +89,35 @@ def decorated(*args: Any, **kwargs: Any) -> None:
                 return func(*args, **kwargs)
 
         return decorated
+
+
+class retry_on_database_error(object):
+
+    def __init__(self, func: Callable) -> None:
+        self.func = func
+
+    def __call__(self, *args: Any, **kwargs: Any) -> Any:
+        self._initialize_retry_parameters()
+        while True:
+            try:
+                result = self.func(*args, **kwargs)
+                break
+            except IntegrityError:
+                self._validate_if_max_number_of_tries_exceeded()
```

Comment:
please rename to `validate_number_of_retries`

Reply:
I renamed to `_validate_number_of_retries`, because this is "private" method and should not be called from outside. Is it OK or it should be just `validate_number_of_retries`? What do You think?